src/main/resources/EDIFACT-SupplyChain-D03B/EDIFACT-SupplyChain-Messages-D.03B.xsd

```diff
@@ -87,99 +87,82 @@
         <xsd:element dfdl:ref="ibmEdiFmt:EDISegmentFormat" dfdl:initiator="UNH" name="UNH" type="srv:UNH-MessageHeader"/>
         <xsd:choice>
           <xsd:sequence>
-            <xsd:choice>
-              <xsd:element ref="srv:AUTACK">
+            <xsd:choice dfdl:choiceDispatchKey=
+              "{ 
+                 if (fn:contains(./UNH/S009/E0065,'AUTACK')) then 'autack'
```

Comment:
Hmm. Why not just have the expression be { ./UNH/S009/E0065 } so that we avoid this linear chain of if-then-else logic? The branch keys would become the upper-case variants. The badmsg has to be handled in a separate choice that surrounds this choice-by-dispatch.